trieste/models/gpflow/sampler.py

```diff
@@ -0,0 +1,350 @@
+# limitations under the License.
+"""
+This module is the home of the sampling functionality required by Trieste's
+GPflow wrappers.
+"""
+
+from __future__ import annotations
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+from gpflux.layers.basis_functions import RandomFourierFeatures
+
+from ...data import Dataset
+from ...types import TensorType
+from ...utils import DEFAULTS
+from ..interfaces import (
+    ProbabilisticModel,
+    ReparametrizationSampler,
+    TrajectoryFunction,
+    TrajectorySampler,
+)
+
+
+class IndependentReparametrizationSampler(ReparametrizationSampler):
+    r"""
+    This sampler employs the *reparameterization trick* to approximate samples from a
+    :class:`ProbabilisticModel`\ 's predictive distribution as
+
+    .. math:: x \mapsto \mu(x) + \epsilon \sigma(x)
+
+    where :math:`\epsilon \sim \mathcal N (0, 1)` is constant for a given sampler, thus ensuring
+    samples form a continuous curve.
+    """
+
+    def __init__(self, sample_size: int, model: ProbabilisticModel):
+        """
+        :param sample_size: The number of samples to take at each point. Must be positive.
+        :param model: The model to sample from.
+        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive.
+        """
+        super().__init__(sample_size, model)
+
+        # _eps is essentially a lazy constant. It is declared and assigned an empty tensor here, and
+        # populated on the first call to sample
+        self._eps = tf.Variable(
+            tf.ones([sample_size, 0], dtype=tf.float64), shape=[sample_size, None]
+        )  # [S, 0]
+
+        self._initialized = tf.Variable(False)
+
+    def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorType:
+        """
+        Return approximate samples from the `model` specified at :meth:`__init__`. Multiple calls to
+        :meth:`sample`, for any given :class:`IndependentReparametrizationSampler` and ``at``, will
+        produce the exact same samples. Calls to :meth:`sample` on *different*
+        :class:`IndependentReparametrizationSampler` instances will produce different samples.
+
+        :param at: Where to sample the predictive distribution, with shape `[..., 1, D]`, for points
+            of dimension `D`.
+        :param jitter: The size of the jitter to use when stabilising the Cholesky decomposition of
+            the covariance matrix.
+        :return: The samples, of shape `[..., S, 1, L]`, where `S` is the `sample_size` and `L` is
+            the number of latent model dimensions.
+        :raise ValueError (or InvalidArgumentError): If ``at`` has an invalid shape or ``jitter``
+            is negative.
+        """
+        tf.debugging.assert_shapes([(at, [..., 1, None])])
+        tf.debugging.assert_greater_equal(jitter, 0.0)
+
+        mean, var = self._model.predict(at[..., None, :, :])  # [..., 1, 1, L], [..., 1, 1, L]
+        var = var + jitter
+
```

Comment:
Why would this need jitter?

Reply:
I added this in, so we could have a single ReparameterizationClass. I agree that it doesnt make sense. Any way round this @uri-granta ?

Reply:
Removed jitter for indep